src/client/components/dimension-list-tile/dimension-list-tile.tsx

```diff
@@ -146,58 +146,29 @@ export class DimensionListTile extends React.Component<DimensionListTileProps, D
     />;
   }
 
-  render() {
-    var { essence, style } = this.props;
-    var { menuDimension, highlightDimension, showSearch, searchText } = this.state;
-    var { dataCube } = essence;
-    var shownDimensions = dataCube.dimensions.toArray();
-    var itemY = 0;
-
-    if (searchText) {
-      shownDimensions = shownDimensions.filter((r) => {
-        return r.title.toLowerCase().indexOf(searchText.toLowerCase()) !== -1;
-      });
-    }
+  private renderMessageIfNoDimensionsFound(dimensionsForView: DimensionOrGroupForView[]) {
+    const { searchText } = this.state;
 
-    const dimensionItems = shownDimensions.map((dimension) => {
-      var className = classNames(
-        DIMENSION_CLASS_NAME,
-        'type-' + dimension.className,
-        {
-          highlight: dimension === highlightDimension,
-          selected: dimension === menuDimension
-        }
-      );
-      var style = transformStyle(0, itemY);
-      itemY += DIMENSION_HEIGHT;
-
-      return <div
-        className={className}
-        key={dimension.name}
-        onClick={this.clickDimension.bind(this, dimension)}
-        onMouseOver={this.onMouseOver.bind(this, dimension)}
-        onMouseLeave={this.onMouseLeave.bind(this, dimension)}
-        draggable={true}
-        onDragStart={this.dragStart.bind(this, dimension)}
-        style={style}
-      >
-        <div className="icon">
-          <SvgIcon svg={require('../../icons/dim-' + dimension.className + '.svg')}/>
-        </div>
-        <div className="item-title">
-          <HighlightString className="label" text={dimension.title} highlight={searchText} />
-        </div>
-
-      </div>;
-    }, this);
-
-    var message: JSX.Element = null;
-    if (searchText && !dimensionItems.length) {
-      message = <div className="message">{`No ${ STRINGS.dimensions.toLowerCase() } for "${searchText}"`}</div>;
+    if (!!searchText && !dimensionsForView.some(dimension => dimension.hasSearchText)) {
+      return <div className="message">{`No ${ STRINGS.dimensions.toLowerCase() } for "${searchText}"`}</div>;
+    } else {
+      return <></>;
     }
+  }
+
+  render() {
+    const { essence, style } = this.props;
+    const { menuDimension, showSearch, searchText } = this.state;
+    const { dataCube } = essence;
+
+    const dimensionsConverter = new DimensionsConverter(hasSearchTextPredicate(searchText), isFilteredOrSplitPredicate(essence), menuDimension);
```

Comment:
No need for classes - it's creating new instance on each render (even with no actual view difference).